google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminClient.java

```diff
@@ -327,18 +329,68 @@ public Table getTable(String tableId) {
    * }</pre>
    */
   public ApiFuture<List<TableName>> listTablesAsync() {
-    ApiFuture<ListTablesPagedResponse> listResp =
-        this.stub.listTablesPagedCallable().futureCall(composeListTableRequest());
+    ListTablesRequest request = ListTablesRequest.newBuilder().setParent(instanceName.toString())
+        .build();
 
-    return ApiFutures.transform(
-        listResp,
-        new ApiFunction<ListTablesPagedResponse, List<TableName>>() {
+    // TODO(igorbernstein2): try to upstream pagination spooling or figure out a way to expose the
+    // paginated responses while maintaining the wrapper facade.
+
+    // Fetch the first page.
+    ApiFuture<ListTablesPage> firstPageFuture = ApiFutures.transform(
+        stub.listTablesPagedCallable().futureCall(request),
+        new ApiFunction<ListTablesPagedResponse, ListTablesPage>() {
           @Override
-          public List<TableName> apply(ListTablesPagedResponse input) {
-            return convertToTableNames(input.iterateAll());
+          public ListTablesPage apply(ListTablesPagedResponse response) {
+            return response.getPage();
           }
         },
-        MoreExecutors.directExecutor());
+        MoreExecutors.directExecutor()
+    );
+
+    // Fetch the rest of the pages by chaining the futures.
+    ApiFuture<List<com.google.bigtable.admin.v2.Table>> allProtos = ApiFutures
```

Comment:
nit: "ApiFuture<List<com.google.bigtable.admin.v2.Table>> allTables"?

Reply:
I think keeping the unwrapped objects named protos makes it easier to follow